tests/workflows/test_release_workflow_healing.py

```
@@ -147,3 +148,72 @@ def test_release_publish_tolerates_partial_build_failures() -> None:
         assert step.get("continue-on-error") is True, (
             f"'{step_name}' must tolerate absent artefacts from failed legs"
         )
+
+
+def _step_names(workflow_path, job_name):
```

Ruff (0.16.3)

[warning] Missing return type annotation for private function `_step_names`

(ANN202)